integration-tests/search.test.ts

```diff
@@ -161,6 +162,99 @@ describe('search', () => {
         );
     });
 
+    test('quantity', async () => {
+        const observation = {
+            resourceType: 'Observation',
+            status: 'final',
+            code: {
+                coding: [
+                    {
+                        system: 'http://loinc.org',
+                        code: '29463-7',
+                        display: 'Body Weight',
+                    },
+                ],
+            },
+            valueQuantity: {
+                value: 185,
+                unit: 'lbs',
+                system: 'http://unitsofmeasure.org',
+                code: '[lb_av]',
+            },
+        };
+
+        const testObservation = (await client.post('Observation', observation)).data;
+        await waitForResourceToBeSearchable(client, testObservation);
+
+        const aFewMinutesAgo = aFewMinutesAgoAsDate();
+
+        const testsParams = [
+            { 'value-quantity': '185|http://unitsofmeasure.org|[lb_av]' },
+            { 'value-quantity': '185||[lb_av]' },
+            { 'value-quantity': '185' },
+            { 'value-quantity': 'ge185' },
+            { 'value-quantity': 'le185' },
+            { 'value-quantity': 'gt184.5' },
+            { 'value-quantity': 'sa184.5' },
+            { 'value-quantity': 'lt200' },
+            { 'value-quantity': 'eb200' },
+            { 'value-quantity': 'eq1.8e2' },
+        ].map(params => ({
+            url: 'Observation',
+            params: { _lastUpdated: `ge${aFewMinutesAgo}`, ...params },
+        }));
+
+        // eslint-disable-next-line no-restricted-syntax
+        for (const testParams of testsParams) {
+            // eslint-disable-next-line no-await-in-loop
+            await expectResourceToBePartOfSearchResults(client, testParams, testObservation);
+        }
+    });
+
+    test('numeric', async () => {
+        const chargeItem = {
+            resourceType: 'ChargeItem',
+            status: 'billable',
+            code: {
+                coding: [
+                    {
+                        code: '01510',
+                        display: 'Zusatzpauschale für Beobachtung nach diagnostischer Koronarangiografie',
+                    },
+                ],
+            },
+            subject: {
+                reference: 'Patient/example',
+            },
+            factorOverride: 0.8,
+        };
+
+        const testChargeItem = (await client.post('ChargeItem', chargeItem)).data;
+        await waitForResourceToBeSearchable(client, testChargeItem);
+
+        const aFewMinutesAgo = aFewMinutesAgoAsDate();
+
+        const testsParams = [
+            { 'factor-override': '0.8' },
```

Comment:
Is there a use case for quantity being off by a super small amount? 0.0000000005 or something

I thought I remember seeing it in the search code

Reply:
I think you are referring to the comment: https://github.com/awslabs/fhir-works-on-aws-search-es/pull/58#discussion_r615175029. The trailing decimals are rounding errors.

FHIR does equality search by searching a range around the number to work around the possible rounding errors in float numbers. i.e. `eq0.8` or simply `0.8` is effectively searching the range `[0.75, 0.85]` this allows it to match things like `0.8000005`.

There is one test for the range behavior: `eq1.8e2` which searches for the range `[175-180]`, thus matching `185`

Reply:
Ahh okay thank you